### DISCUSSION TITLE
How is server reboot handled?

### DISCUSSION THREAD

**Discussion by Fire-Dragon-DoL** [basecamp/kamal#369] [Q&A] [answered]
I couldn't find any information about this:
Is the app restarted on server restart?
If yes how is that handled? Is it using systemd unit files?

**Comment by intrip** [comment]
The app is not automatically restarted on server restarts, you should run a `mrsk deploy` after a server restart to handle it.

**Reply by Fire-Dragon-DoL** [reply]
Ok, thank you.
Is there any work toward adding this functionality by any chance?

**Comment by igor-alexandrov** [comment, marked as answer]
@intrip seems like you are wrong. All containers managed with MRSK are started with `--restart unless-stopped` option, which guarantees that they will be restarted always except to situations when they are manually stopped.

The only thing that @Fire-Dragon-DoL should take care about is to ensure that Docker service will be started after reboot. On Ubuntu you can check it with `systemctl list-unit-files | grep docker`. If `docker.service` is disabled, you can enable it with `systemctl enable docker.service`.

**Reply by Fire-Dragon-DoL** [reply]
Would that survive even a server restart? The containers with `--restart unless-stopped` I mean, `systemctl` will definitely restart docker if enabled

**Reply by intrip** [reply]
Interesting to see if `--restart unless-stopped` works through server reboot. I thought it wouldn't work, please let me know.

**Reply by igor-alexandrov** [reply]
@intrip I've ran a test on a weekend (switched off and on the server) and everything works fine.

**Reply by Fire-Dragon-DoL** [reply]
Amazing! Then I can definitely use mrsk. With a combo of MRSK + managed db + managed object storage, I can make operations very simple

**Comment by igor-alexandrov** [comment]
Yes, if you have your docker service controlled by systemctl.Sent from my iPhoneOn 22 Jul 2023, at 20:52, Francesco Belladonna ***@***.***> wrote:﻿
Would that survive even a server restart?

**Comment by neverstew** [comment]
@Fire-Dragon-DoL - good question and really helpful for me as well. I nearly quit early though because the marked answer is not the same as the right answer (which is much better!) Could you update it when you get a chance?

**Reply by Fire-Dragon-DoL** [reply]
Good point, thank you for calling that out, I updated the marked answer

**Comment by sapandiwakar** [comment]
Just a heads up if you encounter containers deployed with Kamal not starting up after a system reboot - we recently solved this issue in our environment.

The problem was related to our logging configuration:

```yml
logging:
  driver: fluentd
```

We were starting fluent-bit as an accessory container with Kamal, which created a timing problem. After a reboot, some Docker containers would try to start before the fluent-bit container was ready, causing them to fail because the logging driver wasn't available. Unfortunately, Docker doesn't automatically retry starting these containers once the logging driver becomes available.

Here's what the error looked like in our Docker logs:

```
Mar 05 02:22:47 signage-server dockerd[1016]: time="2025-03-05T02:22:47.799984328Z" level=error msg="failed to start container" container=3484c83afa68878e2dcfb2113667e6003cc16522236c9117c7844e9711790510 error="failed to create task for container: failed to initialize logging driver: dial tcp 127.0.0.1:24224: connect: connection refused"
```

To solve this, we removed the containerized fluent-bit and installed it as a system service instead.

**Reply by Fire-Dragon-DoL** [reply]
Notice that in docker you can request to wait for a dependency to be _healthy_ before starting the other containers, which would solve your concurrency issue without resorting to installing the service as systemd
